### ISSUE TITLE
email reminders for action-item assignees 3 business days after retro-close

### ISSUE THREAD

**Issue by vanderhoop** [stride-nyc/remote_retro#521]
As a retro participant who was assigned one or multiple action items,
I want to be reminded of my action item
So that I don't forget to complete it.

__By way of explanation:__ our users are busy, and unless they or their teammates are diligent, they stand to forget about their retro action item over the course of a given iteration. This story ensures they have a reminder of their assigned items, and as a nice side effect, also provides a helpful reminder to first-time users that they checked out the app.

__Acceptance Criteria:__

- [ ] action item assignees are emailed:
  - [ ] once and only once...
  - [ ] 3 *business days* after the close of the retro, surfacing, with the following copy:

```
Hi [FIRST_NAME],

Quick reminder that you had the following action-item(s) assigned to you coming out of the retro held on [DATE_OF_RETRO_UPDATED_AT_TIMESTAMP]:

  - take milk and sugar to Bobby's house
  - add pairing matrix to team room

Looking forward,
The RemoteRetro Team
```

__Helpful Links:__

- [Recurring Elixir Jobs](https://dockyard.com/blog/2017/11/29/need-an-elixir-dependency-to-manage-recurring-jobs-not-so-fast)
  - __Note:__ we used a distributed cluster of nodes (currently, 2 nodes) in production, so the implementer should be aware that any code they run could potentially be run concurrently by 2 separate nodes. 
- [Bamboo Hex docs](https://hexdocs.pm/bamboo/1.2.0/readme.html)

**Comment by vanderhoop** [comment]
adding blocked to this until we figure out what conditions constitute whether or not we should send this. It should be 3 days after the retro close, and the retro should have at least 3 users, so that we don't send first-time users who go through their first retro alone and who submit "dkdkdkdkd" as an action item", as that's a silly email to receive.

**Comment by vanderhoop** [comment]
Update:  it's fine. if folks due that, it'll just mean they're exposed to a helpful feature that will remind them that they tried the app out.

**Comment by vanderhoop** [comment]
Note, we're on a pre-v1 version of Bamboo. We might want to upgrade to that version before taking this on, as it'll just make the upgrade harder down the road.

**Comment by vanderhoop** [comment]
__Update:__ I just upgraded us from 0.8 -> 1.2, so we're using the latest. Updated the issue links

**Comment by vanderhoop** [comment]
blocked until we figure out what the scheduling acceptance criteria are, as Bamboo's sendgrid adapter doesn't currently allow us to use the `send_at` parameter

**Comment by codyduval** [comment]
When working at $ecommerce_company, we used Sidekiq's `perform_at` feature to queue up emails for future delivery.  We bumped into all sorts of not-so-edge-casey cases where developers had to be super duper careful not to affect code that would impact queued emails.

My experiences mirror this fellow's: http://lanceolsen.net/never-schedule-future-jobs/

Have you considered some sort of polling solution? Say, an OTP process that queries once a day for all completed retros x days ago?  Also, is this a feature that users opt into (" [ ] Remind me in x days"), or does it just happen?  Folks get crazy protective of their inbox.
